src/Traits/ManagesOneOffPayments.php

```diff
@@ -0,0 +1,125 @@
+        ];
+
+        $attributes = array_merge($defaultOptions, $overrides, [
+            'owner_type' => $this->getMorphClass(),
+            'owner_id' => $this->getKey(),
+            'order_id' => null,
+        ]);
+
+        $attributes['currency'] = Str::upper($attributes['currency']);
+
+        return OrderItem::create($attributes);
+    }
+
+    /**
+     * Invoice the customer for the given amount and generate an invoice immediately.
+     *
+     * @param string $description
+     * @param int $amount
+     * @param array $tabOptions
+     * @param array $paymentOptions
+     * @return \Laravel\Cashier\Order\Order|bool
+     */
+    public function invoiceFor($description, $amount, array $tabOptions = [], $paymentOptions = [])
+    {
+        if ($tabOptions['currency'] ?? false) {
+            $tabOptions['currency'] = Str::upper($tabOptions['currency']);
+            $paymentOptions['currency'] = $tabOptions['currency'];
+        }
+
+        $this->tab($description, $amount, $tabOptions);
+
+        // Force the invoice method to use the same currency as the order item.
+        return $this->invoice($paymentOptions);
+    }
+
+    /**
+     * Invoice the billable entity outside of the regular billing cycle.
+     *
+     * @param array $paymentOptions
+     * @return \Laravel\Cashier\Order\Order|\Laravel\Cashier\SubscriptionBuilder\RedirectToCheckoutResponse
+     */
+    public function invoice(array $paymentOptions = [])
+    {
+        // Normalize currency; set to default if it's missing, capitalize it.
+        if (! ($paymentOptions['currency'] ?? false)) {
+            $paymentOptions['currency'] = Cashier::usesCurrency();
+        }
+        $paymentOptions['currency'] = Str::upper($paymentOptions['currency']);
+
+        // Check if there's something to invoice
+        $itemsToOrder = OrderItem::shouldProcess()
+            ->forOwner($this)
+            ->ofCurrency($paymentOptions['currency'])
+            ->get();
+
+        // No open order items for this user with the specified currency
+        if ($itemsToOrder->isEmpty()) {
+            return false;
+        }
+
+        if ($this->validateMollieMandate()) {
+            return Order::createFromItems($itemsToOrder, [
+                'currency' => $paymentOptions['currency'],
+            ])->processPayment();
+        }
+
+        return $this->newOneOffPaymentViaCheckout($itemsToOrder, $paymentOptions);
+    }
+
+    /**
+     * Create a new RedirectToCheckoutResponse for a one off payment.
+     *
+     * @link https://docs.mollie.com/reference/v2/payments-api/create-payment#parameters
+     * @param OrderItemCollection $items
+     * @param array $oneOffPaymentOptions !Overrides the Mollie payment options
+     * @return RedirectToCheckoutResponse
+     */
+    protected function newOneOffPaymentViaCheckout(OrderItemCollection $items, array $oneOffPaymentOptions = [])
```

Comment:
If we are going to use actions under the hood, it still makes sense to keep a simple OrderItemCollection here.